src/ray/raylet/scheduling/cluster_resource_data.h

```diff
@@ -363,36 +286,28 @@ class TaskResourceInstances {
   /// check fail will occur.
   /// TODO(hchen): We should hide this method, and encapsulate all mutation operations.
   std::vector<FixedPoint> &GetMutable(const ResourceID resource_id) {
-    auto ptr = GetPointer(resource_id);
-    RAY_CHECK(ptr != nullptr) << "Resource ID not found " << resource_id;
-    return *ptr;
+    auto it = resources_.find(resource_id);
+    RAY_CHECK(it != resources_.end()) << "Resource ID not found " << resource_id;
+    return it->second;
   }
 
   /// Check whether a particular resource exists.
-  bool Has(ResourceID resource_id) const {
-    auto ptr = GetPointer(resource_id);
-    return ptr != nullptr && ptr->size() > 0;
-  }
+  bool Has(ResourceID resource_id) const { return resources_.contains(resource_id); }
 
   /// Set the per-instance values for a particular resource.
   TaskResourceInstances &Set(const ResourceID resource_id,
                              const std::vector<FixedPoint> &instances) {
     if (instances.size() == 0) {
       Remove(resource_id);
     } else {
-      auto ptr = GetPointer(resource_id);
-      if (ptr != nullptr) {
-        *ptr = instances;
-      } else {
-        custom_resources_.emplace(resource_id.ToInt(), instances);
```

Comment:
@scv119 I just found a bug introduced by last PR. 
The impact is that , for custom resources, only the first Set will work.
If you see any related issue on the nightly tests, this PR can fix the issue.